drivers/watchdog/wdt_mcux_wwdt.c

```diff
@@ -107,31 +108,37 @@ static int mcux_wwdt_install_timeout(const struct device *dev,
 	data->wwdt_config.timeoutValue =
 		MSEC_TO_WWDT_TICKS(clock_freq, cfg->window.max);
 
-	if (cfg->window.min) {
-		data->wwdt_config.windowValue =
-			MSEC_TO_WWDT_TICKS(clock_freq, cfg->window.min);
-	}
-
-	if ((data->wwdt_config.timeoutValue < MIN_TIMEOUT) ||
-	    ((data->wwdt_config.windowValue != 0xFFFFFFU) &&
-	     (data->wwdt_config.timeoutValue <
-	      data->wwdt_config.windowValue))) {
+	if (data->wwdt_config.timeoutValue > MAX_TIMEOUT ||
+	    data->wwdt_config.timeoutValue < MIN_TIMEOUT) {
+		LOG_ERR("Timeout value out of range");
 		return -EINVAL;
 	}
 
+	/*
+	 * WWDT uses a count-down counter.
+	 * Window value specifies the highest timer value in which a watchdog
+	 * feed can occur. Therefore, calculate the window value as:
+	 * windowValue = timeoutValue - min_window_ticks (maybe 0)
+	 */
+	data->wwdt_config.windowValue =
+		data->wwdt_config.timeoutValue - MSEC_TO_WWDT_TICKS(clock_freq, cfg->window.min);
+
 	if (cfg->flags & WDT_FLAG_RESET_SOC) {
 		data->wwdt_config.enableWatchdogReset = true;
 		LOG_DBG("Enabling SoC reset");
 	}
 
-	if (cfg->callback && (CONFIG_WDT_MCUX_WWDT_WARNING_INTERRUPT_CFG > 0)) {
-		data->callback = cfg->callback;
-		data->wwdt_config.warningValue = CONFIG_WDT_MCUX_WWDT_WARNING_INTERRUPT_CFG;
-	} else if (cfg->callback) {
-		return -ENOTSUP;
+	if (cfg->callback) {
+		if (CONFIG_WDT_MCUX_WWDT_WARNING_INTERRUPT_CFG > 0) {
+			data->callback = cfg->callback;
+			data->wwdt_config.warningValue = CONFIG_WDT_MCUX_WWDT_WARNING_INTERRUPT_CFG;
+		} else {
+			LOG_ERR("Warning interrupt callback requires "
+				"CONFIG_WDT_MCUX_WWDT_WARNING_INTERRUPT_CFG > 0");
+			return -ENOTSUP;
+		}
```

Comment:
suggestions:
1. Seems the callback change is not related with the WWDT window improvement, suggest use a separate patch.
2. Add some comment to explain why do like this.